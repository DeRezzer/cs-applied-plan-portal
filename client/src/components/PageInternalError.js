import React from "react";

class PageInternalError extends React.Component {
  render() {
    return (

      <div className="status-container">
        <h1 className="status-code">500 Internal Server Error</h1>
        <p className="status-text">The server encountered an internal error
          and was unable to complete your request.</p>
      </div>

    );
  }
}
export default PageInternalError;