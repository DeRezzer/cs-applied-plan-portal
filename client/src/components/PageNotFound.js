/** @jsx jsx */

import {css, jsx} from "@emotion/core";

function PageNotFound() {

  const styleStatusContainer = css`
    margin: 100px;
  `;

  const styleStatusCode = css`
  `;

  const styleStatusText = css`
  `;

  return (

    <div className="status-container" css={styleStatusContainer}>
      <h1 className="status-code" css={styleStatusCode}>404 Page Not Found</h1>
      <p className="status-text" css={styleStatusText}>The page you are looking for does not exist.</p>
    </div>

  );
}
export default PageNotFound;