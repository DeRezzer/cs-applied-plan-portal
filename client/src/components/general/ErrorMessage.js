/** @jsx jsx */

import {css, jsx} from "@emotion/core";
import PropTypes from "prop-types";

function ErrorMessage(props) {

  const style = css`

    .error-message, .hidden-error-message {
      margin: 5px;
      padding: 5px;
      border: 1px solid var(--color-red-200);;
      background: var(--color-red-100);
      color: black;
    }

    .hidden-error-message {
      display: none;
    }

  `;

  // When there is no text the error message still takes up the same space.
  // This is to prevent the page from shifting around when an error is displayed.
  return (
    (props.text.length ? (

      <div css={style}>
        <div className="error-message" css={style}>
          {props.text}
        </div>
      </div>

    ) : (

      <div css={style}>
        <div className="hidden-error-message" css={style}>
          No Error
        </div>
      </div>

    ))
  );
}
export default ErrorMessage;

ErrorMessage.propTypes = {
  text: PropTypes.string
};