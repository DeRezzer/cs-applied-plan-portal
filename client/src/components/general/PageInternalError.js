/** @jsx jsx */

import {css, jsx} from "@emotion/core";

function PageInternalError() {

  const styleStatusContainer = css`
    margin: 100px;
  `;

  const styleStatusCode = css`
  `;

  const styleStatusText = css`
  `;

  return (

    <div className="status-container" css={styleStatusContainer}>
      <h1 className="status-code" css={styleStatusCode}>500 Internal Server Error</h1>
      <p className="status-text" css={styleStatusText}>The server encountered an internal error
            and was unable to complete your request.</p>
    </div>

  );
}
export default PageInternalError;