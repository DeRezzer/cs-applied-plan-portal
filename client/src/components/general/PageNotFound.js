/** @jsx jsx */

import {css, jsx} from "@emotion/core";

function PageNotFound() {

  const style = css`
    margin: 100px;
  `;

  return (

    <div className="status-container" css={style}>
      <h1 className="status-code">404 Page Not Found</h1>
      <p className="status-text">The page you are looking for does not exist.</p>
    </div>

  );
}
export default PageNotFound;