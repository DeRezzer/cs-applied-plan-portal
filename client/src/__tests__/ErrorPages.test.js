/* eslint-disable no-undef */
import React from "react";
import {MemoryRouter} from "react-router-dom";
import renderer from "react-test-renderer";

import ErrorMessage from "../components/general/ErrorMessage";
import PageInternalError from "../components/general/PageInternalError";
import PageNotFound from "../components/general/PageNotFound";

describe("Error messages", () => {

  it("renders error message successfully", () => {
    const wrapper = renderer.create(
      <MemoryRouter>
        <ErrorMessage text={"Some error occurred."} />
      </MemoryRouter>).toJSON();
    expect(wrapper).toMatchSnapshot();
  });

  it("renders page internal error successfully", () => {
    const wrapper = renderer.create(
      <MemoryRouter initialEntries={["/500"]}>
        <PageInternalError />
      </MemoryRouter>).toJSON();
    expect(wrapper).toMatchSnapshot();
  });

  it("renders page not found successfully", () => {
    const wrapper = renderer.create(
      <MemoryRouter>
        <PageNotFound />
      </MemoryRouter>).toJSON();
    expect(wrapper).toMatchSnapshot();
  });
});