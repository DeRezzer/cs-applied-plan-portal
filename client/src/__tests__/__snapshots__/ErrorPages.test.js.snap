// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error messages renders error message successfully 1`] = `
<div
  className=" css-fzm5ub-ErrorMessage"
>
  <div
    className="error-message css-fzm5ub-ErrorMessage"
  >
    <p>
      Some error occurred.
    </p>
  </div>
</div>
`;

exports[`Error messages renders page internal error successfully 1`] = `
<div
  className="status-container css-xdwhvj-PageInternalError"
>
  <h1
    className="status-code"
  >
    500 Internal Server Error
  </h1>
  <p
    className="status-text"
  >
    The server encountered an internal error and was unable to complete your request.
  </p>
</div>
`;

exports[`Error messages renders page not found successfully 1`] = `
<div
  className="status-container css-8zetg9-PageNotFound"
>
  <h1
    className="status-code"
  >
    404 Page Not Found
  </h1>
  <p
    className="status-text"
  >
    The page you are looking for does not exist.
  </p>
</div>
`;
